tooling/noir_js_backend_barretenberg/src/index.ts

```diff
@@ -91,16 +94,8 @@ export class BarretenbergBackend implements Backend {
     const splitIndex = proofWithPublicInputs.length - numBytesInProofWithoutPublicInputs;
 
     const publicInputsConcatenated = proofWithPublicInputs.slice(0, splitIndex);
-
-    const publicInputSize = 32;
-    const publicInputs: Uint8Array[] = [];
-
-    for (let i = 0; i < publicInputsConcatenated.length; i += publicInputSize) {
-      const publicInput = publicInputsConcatenated.slice(i, i + publicInputSize);
-      publicInputs.push(publicInput);
-    }
-
     const proof = proofWithPublicInputs.slice(splitIndex);
+    const publicInputs = deflattenPublicInputs(publicInputsConcatenated, this.acirCircuit.abi);
```

Comment:
This is slightly bad as it requires a "backend" to know about the circuit ABI. We have a path to remove this however as the public inputs are also detailed within the "bytecode" object (as this is a serialized `Circuit` struct).

Reply:
can you create an issue for this?